py/desisim/test/test_templates.py

```diff
@@ -27,160 +27,185 @@ def _check_output_size(self, flux, wave, meta):
-        print('In function test_wd_subtype, seed = {}'.format(self.seed))
-        wd = WD(wave=self.wave, subtype='DA')
-        flux, wave, meta, _ = wd.make_templates(self.nspec, seed=self.seed, nocolorcuts=True)
-        self._check_output_size(flux, wave, meta)
-        np.all(meta['SUBTYPE'] == 'DA')
-
-        wd = WD(wave=self.wave, subtype='DB')
-        flux, wave, meta, _ = wd.make_templates(self.nspec, seed=self.seed, nocolorcuts=True)
-        np.all(meta['SUBTYPE'] == 'DB')
-    
-    @unittest.skipUnless(desi_basis_templates_available, '$DESI_BASIS_TEMPLATES was not detected.')
-    @unittest.expectedFailure
-    def test_wd_subtype_failure(self):
-        '''Test a known failure of specifying the white dwarf subtype.'''
-        print('In function test_wd_subtype_failure, seed = {}'.format(self.seed))
-        wd = WD(wave=self.wave, subtype='DA')
-        flux1, wave1, meta1, _ = wd.make_templates(self.nspec, seed=self.seed, nocolorcuts=True)
-        meta1['SUBTYPE'][0] = 'DB'
-        flux2, wave2, meta2, _ = wd.make_templates(input_meta=meta1)
-        
-    @unittest.skipUnless(desi_basis_templates_available, '$DESI_BASIS_TEMPLATES was not detected.')
-    def test_input_meta(self):
-        '''Test that input meta table option works.'''
-        print('In function test_input_meta, seed = {}'.format(self.seed))
-        for T in [ELG, LRG, BGS, QSO, STAR, MWS_STAR, WD]:
-            Tx = T(wave=self.wave)
-            flux1, wave1, meta1, objmeta1 = Tx.make_templates(self.nspec, seed=self.seed)
-            if 'VDISP' in objmeta1.colnames:
-                vdisp = objmeta1['VDISP'].data
-                flux2, wave2, meta2, objmeta2 = Tx.make_templates(input_meta=meta1, vdisp=vdisp)
-            else:
-                flux2, wave2, meta2, objmeta2 = Tx.make_templates(input_meta=meta1)
-            badkeys = list()
-            for key in meta1.colnames:
-                if key in ('REDSHIFT', 'MAG', 'SEED', 'FLUX_G',
-                           'FLUX_R', 'FLUX_Z', 'FLUX_W1', 'FLUX_W2'):
-                    #- not sure why the tolerances aren't closer
-                    if not np.allclose(meta1[key], meta2[key], rtol=1e-4):
-                        print(meta1['OBJTYPE'][0], key, meta1[key], meta2[key])
-                        badkeys.append(key)
-                else:
-                    if not np.all(meta1[key] == meta2[key]):
-                        badkeys.append(key)
-
-            self.assertEqual(len(badkeys), 0, 'mismatch for spectral type {} in keys {}'.format(meta1['OBJTYPE'][0], badkeys))
-            #if np.all(np.allclose(flux1, flux2, atol=1e-3)) is False:
-            #    import pdb ; pdb.set_trace()
-            self.assertTrue(np.all(np.isclose(flux1, flux2, atol=1e-3)))
-            #self.assertTrue(np.allclose(flux1, flux2, rtol=1e-4))
-            self.assertTrue(np.all(wave1 == wave2))
-
+    #@unittest.skipUnless(desi_basis_templates_available, '$DESI_BASIS_TEMPLATES was not detected.')
```

Comment:
As already noted, you should reactivate these tests.

Reply:
Done in https://github.com/desihub/desisim/pull/559/commits/efca0eb81ff6832478517fd8090b7204a4591615.